### PR TITLE
feat: configurator team, label, permission&member

### PR DESCRIPTION
**Pull request by Tguntenaar**
_Resolves #315_

[PREVIEW](https://near.org/geforcy.near/widget/app?page=admin)
comment down below if I need to add you to the moderators

Acceptance Criteria:

- [x] An "Admin" link is visible in the header navigation, visible and accessible only to DevHub admins (not community admins!)

- [x] When a DevHub admin clicks the Admin button, they see a settings page with the existing section for configuring the featured communities + a new section for configuring the label permissions

- [x] Make the UI more intuitive for label permissions so that it is clear for a moderator how they can perform the common tasks

In my last call with ori about issue #129 we discussed that I should rethink the formally known Teams page into an intuitive UI. I have made two major improvents and some minor:

1. We decided to simplify the task by only allowing 1 label per team. This made possible that the user can create a label and team at the same time instead of seperate and having to link them together afterwards.

2. The next improvement I made is removing most of  the buttons for every action. Before we had a create, an edit and delete button/toggle for every members, labels, teams, permissions. Now this all abstracted away by simply creating, editing and deleting a team in one configurator. 

3. We decided to make a clear distinction between the moderators group and other groups.

Additionally:
The moderators description (in the devhub.near contract) should be changed to:
The moderator group has permissions to edit any posts, and apply all labels including restricted ones.

### CODE REVIEW

File: src/devhub/entity/team/TeamInfo.jsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+  useLabels: uslbls,
+  members: mmbrs,
+}) {
+  let txn = [];
+  let numberOfChanges = 0;
+
+  if (teamName !== tmnm) {
+    numberOfChanges++;
+    if (allTeamNames.includes(`team:${tmnm}`)) {
+      // Error team already exists
```
Comment (Tguntenaar) [inline src/devhub/entity/team/TeamInfo.jsx:70]: Still have to do this error

File: src/devhub/page/admin.jsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+  description,
+  label,
+  editPost,
+  useLabels,
+  members,
+}) {
+  let txn = [];
+
+  if (rootMembers.includes(`team:${teamName}`)) {
+    // Error team already exists
```
Comment (Tguntenaar) [inline src/devhub/page/admin.jsx:100]: Still have to do this as well

File: src/devhub/page/admin.jsx [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    },
+  ]);
+}
+
+const [createTeam, setCreateTeam] = useState(false);
+
+return (
+  <Container>
+    <div className="d-flex flex-column gap-4 p-4">
+      {/* {featuredCommunityList ? (
```
Comment (Tguntenaar) [inline src/devhub/page/admin.jsx:163]: This is issue #275 which I will  follow up after this PR.

**Comment by Tguntenaar** [comment]
Hi @ailisp! Thank you for reviewing. As I mentioned in the issue [here](https://github.com/near/neardevhub-widgets/issues/315#issuecomment-1823557854). I still had to update the preview with the latest changes which I'm planning to do today.

**Comment by Tguntenaar** [comment]
Hi @ailisp! It is ready for review, I have updated the tests and the [preview](https://near.org/geforcy.near/widget/app?page=admin).

**Comment by ailisp** [comment]
@Tguntenaar I tested, UI and the functionalities to edit label permissions look great! There're some bugs:
- After add/remove a featured community, redirect back from wallet, it's succeed but not reflected on the UI. The changes only visible after refresh.
- When I try adding `boatnear.near` as moderator, I got this error:
```
{"index":1,"kind":{"ExecutionError":"Smart contract panicked: panicked at 'Member does not exist', src/access_control/members.rs:201:52"}}
```

**Comment by Tguntenaar** [comment]
@ailisp Hmm that's strange I did push a fix for that exact error. I will look into it and fix it.

**Comment by Tguntenaar** [comment]
@ailisp Hmmm interesting, I did address this specific error in my code, but it seems it can still occur. I'll investigate further and ensure to resolve it by today

**Comment by Tguntenaar** [comment]
@ailisp I did some testing and can reproduce your error. It seems that after a transaction is made, the getAllCommunitiesMetadata function is called 4 times total. First time it returns null, than an array with the length of 4 (old data) and than the updated array with a length of 5 (after the transaction data). However somehow the useState hook trips and the visible data on the page shows the array of 4. I've not been able to resolve this issue, any ideas?

**Comment by Tguntenaar** [comment]
@ailisp Hmm, I might have overwritten the preview with another version. I deployed again and it worked for me. Could you try it once more?

Screenshot 2023-11-30 at 19 15 27

**Comment by ailisp** [comment]
@Tguntenaar I have trouble to test it as currently wallet seems having trouble to connect to RPC nodes (certainly not related to your PR)
image

I'll try again later

**Comment by ailisp** [comment]
Today wallet fails due to network congestion is resolved. This problem also doesn't exist now:
> @ailisp I did some testing and can reproduce your error. It seems that after a transaction is made, the getAllCommunitiesMetadata function is called 4 times total. First time it returns null, than an array with the length of 4 (old data) and than the updated array with a length of 5 (after the transaction data). However somehow the useState hook trips and the visible data on the page shows the array of 4. I've not been able to resolve this issue, any ideas?

Maybe it's also due to network congestion or there was fix on the gateway side

**Pull request merged**